### PR TITLE
Save a draft version of feature settings

### PR DESCRIPTION
**Pull request by felipeelia**
And apply them before a full sync

### Description of the Change

Closes #

### Changelog Entry

> Changed - Depending on the requirements, some feature settings are now saved to be applied after a full sync

### Credits

Props @felipeelia @JakePT 

### Checklist:

- [ ] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

File: includes/classes/REST/Features.php [hunk 1]
```diff
@@ -101,42 +101,84 @@ public function check_permission() {
...
 
-			FeaturesStore::factory()->update_feature( $slug, $settings[ $slug ] );
+		foreach ( $current_settings as $slug => $feature ) {
+			FeaturesStore::factory()->update_feature( $slug, $feature );
 		}
 
-		Utils\update_option( 'ep_feature_settings', $settings );
+		if ( $current_settings !== $new_settings ) {
+			FeaturesStore::factory()->save_feature_settings_draft( $new_settings );
+		}
```
Comment (JakePT) [inline includes/classes/REST/Features.php]: @felipeelia We aren't deleting draft settings or updating them if this condition is false, which means that, for example, if I enable Instant Results and Sync Later, it creates draft settings with Instant Results active. If I then disable Instant Results without syncing the draft settings still exist with Instant Results enabled.

File: includes/classes/Features.php [hunk 2]
```diff
@@ -316,6 +322,51 @@ public function setup_features() {
...
+	 * @since 5.0.0
+	 */
+	public function apply_draft_feature_settings() {
+		$draft_settings = Utils\get_option( 'ep_feature_settings_draft', false );
+		if ( ! $draft_settings ) {
+			return;
+		}
+
+		Utils\update_option( 'ep_feature_settings', $draft_settings );
+		Utils\delete_option( 'ep_feature_settings_draft' );
```
Comment (JakePT) [inline includes/classes/Features.php]: @felipeelia A potential problem here is that this approach means that the draft settings never go through `update_feature`, which means they don't go through the `ep_sanitize_feature_settings` filter or trigger the `post_acivation()` method.

**Review by JakePT — CHANGES_REQUESTED** [review]

File: includes/classes/Features.php (hunk 2 above)
Comment (felipeelia) [reply includes/classes/Features.php]: Changed in [96c160f](https://github.com/10up/ElasticPress/pull/3720/commits/96c160f68434a85bbb7b3d8856986230426b8b07)

File: includes/classes/REST/Features.php (hunk 1 above)
Comment (felipeelia) [reply includes/classes/REST/Features.php]: Changed in [96c160f](https://github.com/10up/ElasticPress/pull/3720/commits/96c160f68434a85bbb7b3d8856986230426b8b07)

**Pull request merged**